src/Control/Distributed/Process/Internal/Primitives.hs

```diff
@@ -982,7 +983,10 @@ expectTimeout n = receiveTimeout n [match return]
 spawnAsync :: NodeId -> Closure (Process ()) -> Process SpawnRef
 spawnAsync nid proc = do
   spawnRef <- getSpawnRef
-  sendCtrlMsg (Just nid) $ Spawn proc spawnRef
+  node     <- getSelfNode
+  if nid == node
+    then sendCtrlMsg Nothing $! Spawn (DS.force proc) spawnRef
```

Comment:
Why special case in this function? Surely the same reasoning applies everywhere else? Can't we push this logic to `sendCtrlMsg`? Further, we *must* be consistent in how we obtain semantic consistency between the remote and the local case: currently for better or worse d-p does not use NFData for this purpose but instead `seq (encode x) x`.

In any case `NFData` is overkill here: `Spawn` is strict in its first argument, and `Closure` doesn't normally embed thunks.